grid2op/Opponent/RandomLineOpponent.py

```diff
@@ -0,0 +1,123 @@
+from grid2op.Opponent import BaseOpponent
+from grid2op.Converter import LineDisconnection
+
+
+class RandomLineOpponent(BaseOpponent):
+    def __init__(self, action_space):
+        # Apply converter
+        converter_action_space = LineDisconnection(action_space)
+        BaseOpponent.__init__(self, converter_action_space)
+        self.action_space.init_converter()
+
+        # Filter lines
+        if action_space.n_line == 59: # WCCI
+            lines_maintenance = ["26_30_56", "30_31_45", "16_18_23", "16_21_27", "9_16_18", "7_9_9",
+                                 "11_12_13", "12_13_14", "2_3_0", "22_26_39" ]
+        elif action_space.n_line == 20: # case 14
+            lines_maintenance = ["1_3_3", "1_4_4", "3_6_15", "9_10_12", "11_12_13", "12_13_14"]
+        else:
+            raise Warning(f'Unknown environment found with {action_space.n_line} lines')
+        self.action_space.filter_lines(lines_maintenance)
+
+        self._do_nothing = self.action_space.actions[0]
+        self._attacks = self.action_space.actions[1:]
+
+    def init(self, *args, **kwargs):
+        """
+        Generic function used to initialize the derived classes. For example, if an opponent reads from a file, the
+        path where is the file is located should be pass with this method.
+        """
+        pass
+
+    def reset(self, initial_budget):
+        """
+        This function is called at the end of an episode, when the episode is over. It aims at resetting the
+        self and prepare it for a new episode.
+
+        Parameters
+        ----------
+        initial_budget: ``float``
+            The initial budget the opponent has
+        """
+        pass
+
+    def attack(self, observation, agent_action, env_action, budget, previous_fails):
+        """
+        This method is the equivalent of "attack" for a regular agent.
+
+        Opponent, in this framework can have more information than a regular agent (in particular it can
+        view time step t+1), it has access to its current budget etc.
+
+        Parameters
+        ----------
+        observation: :class:`grid2op.Observation.Observation`
+            The last observation (at time t)
+
+        opp_reward: ``float``
+            THe opponent "reward" (equivalent to the agent reward, but for the opponent) TODO do i add it back ???
+
+        done: ``bool``
+            Whether the game ended or not TODO do i add it back ???
+
+        agent_action: :class:`grid2op.Action.Action`
+            The action that the agent took
+
+        env_action: :class:`grid2op.Action.Action`
+            The modification that the environment will take.
+
+        budget: ``float``
+            The current remaining budget (if an action is above this budget, it will be replaced by a do nothing.
+
+        previous_fails: ``bool``
+            Wheter the previous attack failed (due to budget or ambiguous action)
+
+        Returns
+        -------
+        attack: :class:`grid2op.Action.Action`
```

Comment:
update the doc, it returns an Action, or None, if the opponent decide not to attack

Reply:
done